protocol/ibd/handle_get_blocks.go

```diff
@@ -0,0 +1,80 @@
+package ibd
+
+import (
+	"github.com/kaspanet/kaspad/blockdag"
+	"github.com/kaspanet/kaspad/netadapter/router"
+	"github.com/kaspanet/kaspad/protocol/common"
+	"github.com/kaspanet/kaspad/util/daghash"
+	"github.com/kaspanet/kaspad/wire"
+)
+
+// HandleGetBlocks handles getBlocks messages
+func HandleGetBlocks(incomingRoute *router.Route, outgoingRoute *router.Route, dag *blockdag.BlockDAG) error {
+	for {
+		lowHash, highHash, shouldContinue, err := receiveGetBlocks(incomingRoute)
+		if err != nil {
+			return err
+		}
+		if !shouldContinue {
+			return nil
+		}
+
+		msgIBDBlocks, err := buildMsgIBDBlocks(lowHash, highHash, dag)
+		if err != nil {
+			return err
+		}
+
+		shouldContinue, err = sendMsgIBDBlocks(outgoingRoute, msgIBDBlocks)
+		if err != nil {
+			return err
+		}
+		if !shouldContinue {
+			return nil
+		}
+	}
+}
+
+func receiveGetBlocks(incomingRoute *router.Route) (lowHash *daghash.Hash,
+	highHash *daghash.Hash, shouldContinue bool, err error) {
+
+	message, isOpen := incomingRoute.Dequeue()
+	if !isOpen {
+		return nil, nil, false, nil
+	}
+	msgGetBlocks := message.(*wire.MsgGetBlocks)
+
+	return msgGetBlocks.LowHash, msgGetBlocks.HighHash, true, nil
+}
+
+func buildMsgIBDBlocks(lowHash *daghash.Hash, highHash *daghash.Hash,
+	dag *blockdag.BlockDAG) ([]*wire.MsgIBDBlock, error) {
+
+	blockHashes, err := dag.AntiPastHashesBetween(lowHash, highHash, wire.MaxInvPerMsg)
+	if err != nil {
+		return nil, err
+	}
+
+	msgIBDBlocks := make([]*wire.MsgIBDBlock, len(blockHashes))
+	for i, blockHash := range blockHashes {
+		block, err := dag.BlockByHash(blockHash)
+		if err != nil {
+			return nil, err
+		}
+		msgIBDBlocks[i] = wire.NewMsgIBDBlock(block.MsgBlock())
+	}
+
+	return msgIBDBlocks, nil
+}
+
+func sendMsgIBDBlocks(outgoingRoute *router.Route, msgIBDBlocks []*wire.MsgIBDBlock) (shouldContinue bool, err error) {
+	for _, msgIBDBlock := range msgIBDBlocks {
+		isOpen, err := outgoingRoute.EnqueueWithTimeout(msgIBDBlock, common.DefaultTimeout)
```

Comment:
This is dangerous, since we are sending here quite a mass of blocks, we run a real chance of clogging the buffer, and the only thing that might save us is that we'll be able to finish all the sending within common.DefaultTimeout.

I don't think we should rely on DefaultTimeout in this case. 
Even more then that - we shouldn't rely on the buffer at all, since other messages might be coming simultaneously from other flows.

Instead, we need some indication that block was actually sent before sending the next one.